WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt

```diff
@@ -0,0 +1,73 @@
+package org.wordpress.android.ui.jetpack.backup
+
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.TEST_DISPATCHER
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.BackupAvailableItemsProvider
+import org.wordpress.android.ui.jetpack.backup.details.BackupDownloadDetailsViewModel
+import org.wordpress.android.ui.jetpack.backup.details.BackupDownloadDetailsViewModel.UiState
+import org.wordpress.android.ui.jetpack.backup.details.BackupDownloadDetailsViewModel.UiState.Content
+
+@InternalCoroutinesApi
+class BackupDownloadDetailsViewModelTest : BaseUnitTest() {
+    private lateinit var viewModel: BackupDownloadDetailsViewModel
+    private lateinit var availableItemsProvider: BackupAvailableItemsProvider
+
+    @Before
+    fun setUp() {
+        availableItemsProvider = BackupAvailableItemsProvider()
+        viewModel = BackupDownloadDetailsViewModel(
+                availableItemsProvider,
+                TEST_DISPATCHER,
+                TEST_DISPATCHER
+        )
+    }
+
+    @Test
+    fun `when available items are fetched, the content view is shown`() = test {
+        val uiStates = initObservers().uiStates
+
+        viewModel.start()
+
+        assertThat(uiStates[0]).isInstanceOf(UiState.Content::class.java)
+    }
+
+    @Test
+    fun `item is unchecked, when it is clicked on`() = test {
+        val uiStates = initObservers().uiStates
+
+        viewModel.start()
+
+        ((uiStates.last() as Content).items[1]).onClick.invoke()
+
+        assertThat(((uiStates.last() as Content).items[1]).checked).isFalse()
+    }
+
+    @Test
+    fun `item is checked, when it is clicked on twice `() = test {
```

Comment:
Idea (💡 ): These kind of tests scenarios can benefit from a `given/when/then` pattern. Utilising this pattern, I would have named this test case as: `given item is unchecked, when item is clicked, then item gets checked`. Also I would have asserted the `unchecked` state in the setup, like with the `.isFalse()` above and then, trigger the next `.invoke()` action separately and check the `checked` state afterwards.